src/Components/RoleToolbar.js

```diff
@@ -0,0 +1,227 @@
+                </DropdownToggle>
+              }
+              dropdownItems={['Role name', 'Application'].map((colName) => (
+                // Filterable columns are RequestID, AccountID, and Status
+                <DropdownItem key={colName} value={colName} component="button">
+                  {capitalize(colName)}
+                </DropdownItem>
+              ))}
+            />
+            {filterColumn === 'Application' ? (
+              <React.Fragment>
+                <span id={selectLabelId} hidden>
+                  {selectPlaceholder}
+                </span>
+                <Select
+                  aria-labelledby={selectLabelId}
+                  variant="checkbox"
+                  aria-label="Select applications"
+                  onToggle={(isOpen) => setIsSelectOpen(isOpen)}
+                  onSelect={(_ev, selection) => {
+                    if (appSelections.includes(selection)) {
+                      setAppSelections(
+                        appSelections.filter((s) => s !== selection)
+                      );
+                    } else {
+                      setAppSelections([...appSelections, selection]);
+                    }
+                  }}
+                  isOpen={isSelectOpen}
+                  selections={appSelections}
+                  isCheckboxSelectionBadgeHidden
+                  placeholderText={selectPlaceholder}
+                  style={{ maxHeight: '400px', overflowY: 'auto' }}
+                >
+                  {applications.map((app) => (
+                    <SelectOption key={app} value={app}>
+                      {capitalize(app.replace(/-/g, ' '))}
+                    </SelectOption>
+                  ))}
+                </Select>
+              </React.Fragment>
+            ) : (
+              <TextInput
+                name="rolesSearch"
+                id="rolesSearch"
+                type="search"
+                iconVariant="search"
+                aria-label="Search input"
+                placeholder="Filter by role name"
+                value={nameFilter}
+                onChange={(val) => setNameFilter(val)}
+              />
+            )}
+          </InputGroup>
+        </ToolbarItem>
+        <ToolbarItem variant="pagination" align={{ default: 'alignRight' }}>
+          <AccessRequestsPagination id="top" />
+        </ToolbarItem>
+      </ToolbarContent>
+      {hasFilters && (
+        <ToolbarContent>
+          {nameFilter && (
+            <ChipGroup categoryName="Role name">
+              <Chip onClick={() => setNameFilter('')}>{nameFilter}</Chip>
+            </ChipGroup>
+          )}
+          {appSelections.length > 0 && (
+            <ChipGroup categoryName="Status">
+              {appSelections.map((status) => (
+                <Chip
+                  key={status}
+                  onClick={() =>
+                    setAppSelections(appSelections.filter((s) => s !== status))
```

Comment:
When computing a new state from the previous state, it's better to use the callback variant of the useState setter to avoid potentially using out of sync values.

There are a few other cases in the code above.

```js
setAppSelections((prev => prev.filter((s) => s !== status))
```